adding-zkpoex-blog (#140)
risc0/website: blog/2023-03-xx-zkpoex/index.md

---
slug: zkpoex
title: 'Revolutionizing DeFi Security: ZK Proof of Exploit on RISC Zero'
authors: [paul]
tags: [zkpoex, ethdenver]
---

We are thrilled to share the recent success of a groundbreaking project, zkPoEX (Zero-Knowledge Proof of Exploit), which won 1st place in the ETHDenver Hackathon DeFi Track. 
[zkPoEX](https://github.com/zkoranges/zkPoEX) enables secure collaboration between security experts and DeFi projects.

> Built using RISC Zero’s zero-knowledge virtual machine (zkVM), zkPoEX is an innovative Proof of Concept (PoC) that showcases a simple reentrancy case; 

Reviewer: "zkPoEX is a Proof of Concept"

Would like to reduce the amount of "marketing" language

> Built using RISC Zero’s zero-knowledge virtual machine (zkVM), zkPoEX is an innovative Proof of Concept (PoC) that showcases a simple reentrancy case; 

Author: ```suggestion
Built using RISC Zero’s zero-knowledge virtual machine (zkVM), zkPoEX is a Proof of Concept (PoC) that showcases a simple reentrancy case; 
```

> it's truly one of the most exciting applications of zero-knowledge technology in the DeFi landscape.

Reviewer: It's an exciting application of zero-knowledge technology in the DeFi landscape

> it's truly one of the most exciting applications of zero-knowledge technology in the DeFi landscape.

Author: ```suggestion
it's an exciting application of zero-knowledge technology in the DeFi landscape.
```

The journal includes committed information, which consists of the hash of the public state, changes made to the state, such as a decrease in the balance of the Target Contract, and a commitment to the private inputs, represented by their hash.

As a result of the versatile design of the zkVM it should be noted that:
State change outputs may be customized or enforced according to specific requirements.
For a more general approach, the Target Contract can be treated as a public input.
The constraint system within the Guest can be made more complex, depending on the specific use case.

Running the PoC required zkPoEX’s team to reduce the number of interactions between the Exploiter and Target Contracts to reduce the amount of computation. 
That led to approximately 16GB RAM requirement and 4 minutes runtime using an Apple M1 Max with 10-core CPU and 32-core GPU.

# zkPoEX and RISC Zero

zkPoEX was developed by [@zkoranges](https://twitter.com/zkoranges) and [@feder.eth](https://twitter.com/feder_eth) with acknowledgements to Daniel Lumi for advisory and Maciej Zieliński for his insightful [blog post](https://odra.dev/blog/evm-at-risc0/) on EVM inside the RISC Zero zkVM. 

> As zkPoEX's success in the ETHDenver hackathon DeFi track demonstrates, there is immense potential for zero-knowledge proof technology in securing the DeFi ecosystem. 

Reviewer: DeFi track demonstrates the potential for zero-knowledge proof ....

> As zkPoEX's success in the ETHDenver hackathon DeFi track demonstrates, there is immense potential for zero-knowledge proof technology in securing the DeFi ecosystem. 

Author: ```suggestion
zkPoEX's success in the ETHDenver hackathon DeFi track demonstrates the potential for zero-knowledge proof technology in securing the DeFi ecosystem. 
```

> The RISC Zero zkVM, with its ability to facilitate the creation and verification of zero-knowledge proofs, is uniquely positioned to power such revolutionary applications like zkPoEX.

Reviewer: "to power innovative applications like zkPoEX"

Author: ```suggestion
The RISC Zero zkVM, with its ability to facilitate the creation and verification of zero-knowledge proofs, is uniquely positioned to power innovative applications like zkPoEX.
```
